### PR TITLE
Feature add breathe and m2r

### PR DESCRIPTION
**Pull request by bvanessen**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-breathe/package.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    version('4.9.1',  sha256='3e1e31b879dcd1fe1ebdaf4d0b11356b8e348ac5af101cafa9e54956716a7f02')
+    version('4.9.0',  sha256='a2b4b8cd2c4ef708ae69dd174e65731b1e18c24f8670036d2f5e608558be2613')
+    version('4.8.0',  sha256='edac7732ad8702ea0425773f1f0c98e7ad5028dbba5fe6483f32a1df3afe2f31')
+    version('4.7.3',  sha256='35e2e937fad97c6d7e287db6007184325284130ab50e1154fe126cffc09a1989')
+    version('4.7.2',  sha256='982d47909d22fcd71b48bad5aef3644294340b24f612b2887cde2e3be464d960')
+    version('4.7.1',  sha256='afb1ab0084b25d3670fa8f5cf2eeaee6fe61bfc77876e3816b140eacd4949875')
+    version('4.7.0',  sha256='5629c67f5adb41f39375d36c5f0d60d34b1230be268125e535205d77f69211e4')
+
+    depends_on('py-setuptools', type='build')
+    depends_on('py-sphinx @1.7.55:',     type=('build', 'run'))
```
[the remark this answers is not in the document]
Comment (bvanessen) [reply var/spack/repos/builtin/packages/py-breathe/package.py]: The main page of documentation says 1.7.5 as the minimum.

File: var/spack/repos/builtin/packages/py-breathe/package.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    depends_on('py-docutils @0.11:',   type=('build', 'run'))
+    depends_on('doxygen @1.8.4:')
```
[the remark this answers is not in the document]
Comment (bvanessen) [reply var/spack/repos/builtin/packages/py-breathe/package.py]: I am happy to add that, but how do you know given that it isn't on their webpage.

File: var/spack/repos/builtin/packages/py-m2r/package.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    version('0.1.14', sha256='d8f844cc6f0c954e55eb3d599e436b337e7f97559dafa7ff54412a5f06f0867c')
+    version('0.1.13', sha256='97f81a45b922f7700e0c0dfd742c79d904e337e92ec4241a11aa579713a1afa4')
+    version('0.1.12', sha256='585b5b4e30494858de49ad844cafeb752bd12764fd6c20ca8496ddd4be5e8d21')
+    version('0.1.11', sha256='a29ef03a0b67116ba8e394a4b3f8afc354211b899f89dedf9c4c0b79828e9ca7')
+    version('0.1.10', sha256='4c7a92f111ac842b9c4d3719b8d785b26cb1cf6e81ec0940f8fe0ccc078e6925')
+    version('0.1.9',  sha256='86fe3c8c54a59807e0ef8cfe5a79fc013b4aaced1f3db493878f5ec1a46bb6d5')
+    version('0.1.8',  sha256='2bfeb2f3de5f9fe9cd411aba82767a6473a43db96ca1730c03b0fe0167239b7e')
+
+    depends_on('py-setuptools', type='build')
+    depends_on('py-mistune',    type=('build', 'run'))
```
[the remark this answers is not in the document]
Comment (bvanessen) [reply var/spack/repos/builtin/packages/py-m2r/package.py]: Okay.

File: var/spack/repos/builtin/packages/py-sphinx/package.py [hunk 4]
```diff
@@ -9,7 +9,7 @@
 class PySphinx(PythonPackage):
     """Sphinx Documentation Generator."""
     homepage = "http://sphinx-doc.org"
-    url      = "https://pypi.io/packages/source/S/Sphinx/Sphinx-1.6.1.tar.gz"
+    url      = "https://github.com/sphinx-doc/sphinx/archive/v1.8.2.tar.gz"
```
[the remark this answers is not in the document]
Comment (bvanessen) [reply var/spack/repos/builtin/packages/py-sphinx/package.py]: Spack checksum id did not work with the existing url.

File: var/spack/repos/builtin/packages/py-breathe/package.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    version('4.9.0',  sha256='a2b4b8cd2c4ef708ae69dd174e65731b1e18c24f8670036d2f5e608558be2613')
+    version('4.8.0',  sha256='edac7732ad8702ea0425773f1f0c98e7ad5028dbba5fe6483f32a1df3afe2f31')
+    version('4.7.3',  sha256='35e2e937fad97c6d7e287db6007184325284130ab50e1154fe126cffc09a1989')
+    version('4.7.2',  sha256='982d47909d22fcd71b48bad5aef3644294340b24f612b2887cde2e3be464d960')
+    version('4.7.1',  sha256='afb1ab0084b25d3670fa8f5cf2eeaee6fe61bfc77876e3816b140eacd4949875')
+    version('4.7.0',  sha256='5629c67f5adb41f39375d36c5f0d60d34b1230be268125e535205d77f69211e4')
+
+    depends_on('py-setuptools', type='build')
+    depends_on('py-sphinx @1.7.55:',     type=('build', 'run'))
+    depends_on('py-docutils @0.11:',   type=('build', 'run'))
```
[the remark this answers is not in the document]
Comment (bvanessen) [reply var/spack/repos/builtin/packages/py-breathe/package.py]: Again this requirement was pulled from the breathe webpage.

File: var/spack/repos/builtin/packages/py-breathe/package.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-breathe/package.py]: Never trust the web page, always trust the `setup.py`

[the remark this answers is not in the document]
Comment (bvanessen) [reply var/spack/repos/builtin/packages/py-breathe/package.py]: Okay, thanks for the tip.

[the remark this answers is not in the document]
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-breathe/package.py]: See https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html for other useful tips.

File: var/spack/repos/builtin/packages/py-sphinx/package.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-sphinx/package.py]: See https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#pypi-vs-github for more discussion on this topic. I won't force you to use PyPI, but if you switch to GitHub, you'll have to update the checksums for all of the old releases.

[the remark this answers is not in the document]
Comment (bvanessen) [reply var/spack/repos/builtin/packages/py-sphinx/package.py]: Then what is the preferred mechanism for getting a checksum when using PyPI?

[the remark this answers is not in the document]
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-sphinx/package.py]: I usually manually download and checksum it myself: `shasum -a 256 foo.tar.gz` on macOS and `sha256sum foo.tar.gz` on Linux. I know, it's a pain in the ass. I've been meaning to hack `spack checksum` to query the PyPI API but they keep changing the API and I haven't figured out how to do it yet.

[the remark this answers is not in the document]
Comment (bvanessen) [reply var/spack/repos/builtin/packages/py-sphinx/package.py]: Okay.  I put in a sha for the most recent version, but I don't have the time to add all of the other ones back in.

**Pull request merged**